crates/node/src/main.rs

```diff
@@ -253,7 +256,10 @@ async fn run(config: Arc<Config>) -> Result<()> {
     let workflow_engine = Arc::new(WorkflowEngine::new(database.clone()));
     let download_url_prefix = format!(
         "http://{}:{}",
-        config.p2p_listen_addr.ip(),
+        config
+            .p2p_advertised_listen_addr
+            .unwrap_or_else(|| p2p_listen_addr)
+            .ip(),
```

Comment:
This accounts for the NAT scenario.

There is still a problem when node is configured to be bound to all interfaces (`0.0.0.0`), because for the peer that address doesn't tell anything. I'm very hesitant on relying on any of public services for finding out the node's public IP so I'm wondering if we could improve the P2P handshake in a way that the peer would report back the public IP and that would then find its way to places where the URLs are constructed?